qiskit/result/distributions/probability.py

```diff
@@ -0,0 +1,26 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Class for probability distributions."""
+
+
+class ProbDistribution(dict):
+    """A generic dict-like class for probability distributions."""
+
+    def __init__(self, data, shots=None):
+        """Builds a probability distribution object.
+
+        Parameters:
+            data (dict): Input probability data.
+            shots (int): Number of shots the distribution was derived from.
+        """
+        self.shots = shots
+        super().__init__(data)
```

Comment:
Do you want want to do any input validation here? Like on the equivalent for counts we check the keys are valid. We also handle various key formats (`int`, hex strings, etc). As it is this is kind of a weird class, it's basically just a dictionary with a `shots` count attribute.

Reply:
At this moment it is not critical as the class instances are just used internally.  However, more robust validation would be useful.  I can think that there would be more to these classes than there is a present.  But this is what is required to move forward with what is required ASAP.

Reply:
I did this in: https://github.com/Qiskit/qiskit-terra/pull/6377/commits/b485ac3f8f880d130bd827fc1bce487773272ca8 let me know if you're ok with these changes.

Reply:
Looks good.  Just a few minor comments, probably just copy/paste errors.